crates/binstalk/src/fetchers/gh_crate_meta/hosting.rs

```diff
@@ -82,10 +87,13 @@ impl RepositoryHost {
     }
 }
 
-fn apply_filenames_to_paths(paths: &[&str], filenames: &[&[&str]]) -> Vec<String> {
+fn apply_filenames_to_paths<'a>(
+    paths: &'a [&'a str],
+    filenames: &'a [&'a [&'a str]],
+) -> impl Iterator<Item = String> + 'a {
```

Comment:
given this is entirely internal to this file and how we're using it, we can just do:

```rust
fn apply_filenames_to_paths(
    paths: &[&str],
    filenames: &[&[&str]],
    suffix: &str,
) -> Vec<String> {
    // ...
    .map(|(filename, path)| format!("{path}/{filename}{suffix}"))
    .collect()
}
```

and save a `format!` as well as saving an iterator.

Reply:
Thanks, that removes all intermediate `String` while improve ergonomic!